runtime/compiler/runtime/RelocationRecord.cpp

```diff
@@ -6366,6 +6369,52 @@ TR_RelocationRecordValidateJ2IThunkFromMethod::methodID(TR_RelocationTarget *rel
    return reloTarget->loadUnsigned16b((uint8_t *) &((TR_RelocationRecordValidateJ2IThunkFromMethodBinaryTemplate *)_record)->_methodID);
    }
 
+
+char *
+TR_RelocationRecordStaticDefaultValueInstance::name()
+   {
+   return "TR_StaticDefaultValueInstance";
+   }
+
+TR_RelocationErrorCode
+TR_RelocationRecordStaticDefaultValueInstance::applyRelocation(TR_RelocationRuntime *reloRuntime, TR_RelocationTarget *reloTarget, uint8_t *reloLocation)
+   {
+   uintptr_t newConstantPool = computeNewConstantPool(reloRuntime, reloTarget, constantPool(reloTarget));
+   TR_OpaqueClassBlock *newClassAddress = computeNewClassAddress(reloRuntime, newConstantPool, inlinedSiteIndex(reloTarget), cpIndex(reloTarget));
+
+   if (!newClassAddress ||
+       !reloRuntime->comp()->fej9()->isClassInitialized(newClassAddress))
+      return TR_RelocationErrorCode::staticDefaultValueInstanceRelocationFailure;
+
+   j9object_t *newDefaultValueSlotAddress = TR::Compiler->cls.getDefaultValueSlotAddress(reloRuntime->comp(), newClassAddress);
+
+   reloTarget->storeAddressSequence((uint8_t *)newDefaultValueSlotAddress, reloLocation, reloFlags(reloTarget));
+
+   RELO_LOG(reloRuntime->reloLogger(), 6, "\tapplyRelocation: newDefaultValueSlotAddress %p\n", newDefaultValueSlotAddress);
+   return TR_RelocationErrorCode::relocationOK;
+   }
+
+TR_RelocationErrorCode
+TR_RelocationRecordStaticDefaultValueInstance::applyRelocation(TR_RelocationRuntime *reloRuntime, TR_RelocationTarget *reloTarget, uint8_t *reloLocationHigh, uint8_t *reloLocationLow)
+   {
+   uintptr_t oldValue = (uintptr_t) reloTarget->loadAddress(reloLocationHigh, reloLocationLow);
+   uintptr_t newConstantPool = computeNewConstantPool(reloRuntime, reloTarget, oldValue);
```

Comment:
```suggestion
   uintptr_t newConstantPool = computeNewConstantPool(reloRuntime, reloTarget, constantPool(reloTarget));
```

Reply:
I know you copied this from `TR_RelocationRecordClassAddress` but I'm pretty sure `TR_RelocationRecordClassAddress` is doing the wrong thing... so, I'll open an issue to track that.

Reply:
Addressed in 38b00cb